Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM ubuntu:20.04 as base
+
+# Config
+ARG BRANCH="master"
+ARG CLANG_VERSION=14
+
+# General dependencies
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update \
+ && apt-get install -yq wget gnupg \
+ && rm -rf /var/lib/apt/lists/*
+
+# Add llvm repo
+RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+RUN echo "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-${CLANG_VERSION} main" >> /etc/apt/sources.list
+
+# Run dependencies
+RUN apt-get update \
+ && apt-get install -yq "clang-${CLANG_VERSION}" \
+ && rm -rf /var/lib/apt/lists/*
+
+
+# Build binder
+FROM base as build
+
+# Build dependencies
+RUN apt-get update
+RUN apt-get install -yq \
+	"libclang-${CLANG_VERSION}-dev" \
+	cmake \
+	git
+
+# Clone binder source
+ARG REPO="https://github.com/RosettaCommons/binder.git"
+RUN git clone --depth 1 --branch "${BRANCH}" "${REPO}" /binder
```

Comment:
Unless i am mistaken using `--depth 1` will lead to issue if user tries to use non-head commits. Also, whole Binder repository is minuscule compare to LLVM clone size. So this option create more of inconvenience than a real space saver, so perhaps this option should removed?

Reply:
There is no following git checkout; since the exact commit being cloned is what is built I have `--depth 1` as it just saves time. I'm not dead set on it, but there should be only disadvantages to removing it unless we also add lines like:
```
ARG COMMIT=HEAD
RUN git checkout "${COMMIT}"
```
Would you prefer those added?

Reply:
I guess my initial comment was a bit cryptic, let me clarify: what i was referring to was the case when someone going to work on Binder development using this image as base for environment. Or for example if someone get into situation when s/he want to try different Binder builds. 

Please note that this comment was optional since we just provide an example of Dockerfile so i am happy to merge this with `depth` option set.

Reply:
I see. Since I'll be moving this into the examples dir I suspect it would be better without that. If you do desire to make a `dev` example, I think you should instead do a `COPY ./binder /binder` so that the devs would not need to pull from a remote source every build (which also means testing small commits would not need to have a git push each time).

Reply:
Hm... i guess `COPY` would make more sense but in the same time using `git` will allow one to use Dockerfile on its own. I think i prefer later a bit more though. Let's accept that things is not 'fully-optimal' here and just merge this, - after all this suppose to. only serve as example on which one could build on.